### DISCUSSION TITLE
Bnx2 devid 14e41639 not included in the bnx rom.

### DISCUSSION THREAD

**Discussion by Phalen** [ipxe/ipxe#302] [General]
I have a couple of older servers that use the bnx based broadcom cards.  they are having issues using ipxe which is built in to openstack to boot.  I went to build a rom for the intel and braodcom the intel built but the broadcom did not.  when i dug around the bnx2.ids.o file did not list the above 3x series but has 4x and Ax in it.  is there a reason it was not included or was it assumed no one uses it any more? would this also be the reason the broadcoms get an ip but fail to configure since the bnx2 does not have info on them when a built ipxe file is used by openstack?

**Comment by NiKiZe** [comment]
Have you tried adding relevant PCIROM line and checked if it works?

**Reply by Phalen** [reply]
grep was marking the .ids.o files as binary so i was not sure if i could safely use vim to add the id to the ids file since im new to ipxe or if they were left out for a specific compatibility reason.

**Reply by NiKiZe** [reply]
https://github.com/ipxe/ipxe/blob/master/src/drivers/net/bnx2.c#L2675 are the lines we are talking about. But be sure it is the right driver.
